tests/js/client/server_parameters/www-authenticate.js

```diff
@@ -62,33 +61,46 @@ function HttpAuthenticateSuite() {
 
     testUnauthorized: function() {
       protocols.forEach((protocol) => {
-        print(`connecting with ${protocol}`);
         connectWith(protocol, user, "");
-        // need to change password, otherwise, the request will return 200
-        users.update(user, "abc");
-        let result = arango.GET_RAW("/_api/version");
-        assertEqual(401, result.code, JSON.stringify(result));
-        assertTrue(result.headers.hasOwnProperty('www-authenticate'), JSON.stringify(result));
-        // to change password back to the original one, we must reconnect with the current one, then change it back to the original one, then connect again
-        arango.reconnectWithNewPassword("abc");
-        users.update(user, "");
-        arango.reconnectWithNewPassword("");
+        let result = "";
+        // we put this in a try to make it sequential, because the request might be made when the password has not
+        // changed yet so it wouldn't get a 401
+        try {
+          // need to change password, otherwise, the request will return 200
+          result = users.update(user, "abc");
+        } finally {
+          assertEqual(result.code, 200, "update password returned with code " + result.code);
```

Comment:
I don't think this is a good solution.
A `finally` block should never have any test assertions.
The purpose of a `finally` block is to clean up after success or failure, no matter what.
It does not make much sense to put assertions into it, because if they fail, the rest of the `finally` block isn't executed and thus the cleanup is not complete. It is like a destructor in C++, which you also don't want to throw its own exceptions.